packages/create-app/src/template/package.json

```
@@ -11,7 +11,7 @@
         "milligram": "^1.4.1",
         "react": "^17.0.1",
         "react-dom": "^17.0.1",
-        "react-scripts": "4.0.3",
+        "react-scripts": "^5.0.0",
```

Unfortunately, I think the actual change is going to need to be a bit bigger. There's two things that the `@datadog/create-app` package does: create an App from an example in the `examples` directory, piece together an App from the template in the `packages/create-app/src/template` directory. The change here only affects the latter use case. Given that your reproduction steps are to use `yarn create @datadog/app`, it looks like you're wanting to do the former use case (which defaults to the `examples/starter-kit` example). In order to address the `yarn create @datadog/app` use case, we'd actually want to update `examples/starter-kit/package.json` in addition to this change.

---

Now the part where it becomes a bigger change than expected. Since we're using yarn classic, and we're using workspaces, and we're using `react-scripts` in most (all?) of the examples, we run into some issues with versions. `react-scripts` wants all of its dependencies to match exactly. If we try to update one example's version of `react-scripts`, it will pull in a different version of dependencies (like `webpack`) and then it will fail with a pretty cryptic error message. My understanding is that yarn classic sometimes hoists dependencies to the root, and `react-scripts` throws an error when there are two different versions somewhere in the dependency chain–even if it _shouldn't_ be able to see it due to them being in different `node_modules` directories.

This problem could probably be solved in a lot of different ways: we could update to yarn berry, but then we run into other dependencies breaking because they don't support yarn berry; we could investigate further about how to prevent hoisting; we could not use `react-scripts` so heavily; we could likely do something else.

The best idea we've come up for this is to update all of the `react-scripts` to the same version. Relatedly, this is also why we're specifying an exact version for `react-scripts`. We didn't do that in the past and when we tried to setup workspaces it all went sideways because one example was using `4.0.x` and another using `4.0.y`. So for the sake of not causing similar issues down the road, we'll want to make sure we use an exact version of the `5.x.y` series.

The reason we haven't done this already is that we haven't looked into how much work it is to do. It might be a one-line change in each of the examples that depend on `react-scripts`. Or, there might be some transitive chain of dependency issues we have to figure out.

---

As far as what to do going forward. It seems like we should update all of the `react-scripts` dependencies to be some exact version of the `5.x.y` series. If this sounds like too much work (since it's unknown what the fallout is) feel free to push back on us to figure it out. I'm more than happy to do this work, but don't want to start on it without checking with you first. If this seems like something you want to do, have at it. It could be really simple! Just don't want you to feel stuck with this ball of yarn because you pulled the thread.

---

It looks like the latest version is `5.0.0`.

```suggestion
        "react-scripts": "5.0.0",
```

---

Thanks for all the context.

I was a little worried that it was going to be a slightly larger issue due to the workspace.

I'll take a swing at getting everything updated and make sure it all still works as expected.